flycheck.el

```diff
@@ -3556,7 +3556,10 @@ The following PROPERTIES constitute an error level:
         (plist-get properties :compilation-level))
   (setf (get level 'flycheck-overlay-category)
         (plist-get properties :overlay-category))
-  (setf (get level 'flycheck-fringe-bitmap-double-arrow)
+  (setf (get level
+        (if (flycheck-high-dpi-screen)
+             'flycheck-fringe-bitmap-double-arrow-hdpi
+             'flycheck-fringe-bitmap-double-arrow))
```

Comment:
This looks good, but for performance reasons we should not call the high-dpi-screen function for each overlay.  Also, the function should likely be called `flycheck--screen-high-dpi-p`, since it's a private (`--`) predicate (`-p`).

Reply:
To clarify, what this code does is attach to the symbol stored in `level` a property named `flycheck-fringe-bitmap-double-arrow`, with value `'flycheck-fringe-bitmap-double-arrow` (easy Lisp symbol has a property list, or plist, essentially a dictionary of static atttributes).  The idea is that users can redefine the bitmap for a given level by changing the value mapped to the key `flycheck-fringe-bitmap-double-arrow` in the level's plist (and the default value happens to be the bitmap called `'flycheck-fringe-bitmap-double-arrow`, and yes it's kind of confusing that they are named the same).

The best would probably be to attach *two* properties, once for the lowdpi marker and one for hidpi one.  Then we can pick when constructing each overlay.

Reply:
@fmdkdd opinions?

Reply:
Yes, adding a new property sounds the right thing to do.  Fringes don't allow anything else other than bitmaps, do they?